### PR TITLE
feat(looker): introduce `build_looker_asset_specs` to replace `@looker_assets`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Model the `dagster-looker` APIs off of https://github.com/dagster-io/dagster/pull/22623 and https://github.com/dagster-io/dagster/pull/22699.

## How I Tested These Changes
pytest

### CODE REVIEW

**Comment by rexledesma** [comment]
* **#22852** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22852?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/views/base_dim_customers.view.lkml
Comment (rexledesma) [inline python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/views/base_dim_customers.view.lkml:1]: This was a duplicate of `customers.view.lkml`, so removing.

File: python_modules/libraries/dagster-looker/dagster_looker/asset_specs.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from .dagster_looker_translator import DagsterLookerTranslator
+
+
+@experimental
+def build_looker_asset_specs(
+    *,
+    project_dir: Path,
+    dagster_looker_translator: Optional[DagsterLookerTranslator] = None,
+) -> Sequence[AssetSpec]:
+    """A decorator for defining Looker assets in a project.
```
Comment (sryza) [inline python_modules/libraries/dagster-looker/dagster_looker/asset_specs.py:21]: No longer a decorator

**Review by sryza — APPROVED** [review]
Nice

**Comment by rexledesma** [comment]
### Merge activity

* **Jul 8, 1:30 PM PDT**: [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/22852) rebased this pull request as part of a merge.

**Pull request merged**